database_cleanup/models/ir_model_fields.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2014-2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, models
+
+
+class IrModelFields(models.Model):
+    _inherit = 'ir.model.fields'
+
+    # In case of purging it means the model does not exist anymore in
+    # installed module. In this specific case, we need to avoid to check
+    # if fields can be removed as it would fail.
+    @api.multi
+    def _prepare_update(self):
+        if self.env.context.get('purge'):
+            not_available_fields = self.browse(False)
+            for record in self:
+                try:
+                    self.env[record.model]
+                except:
```

Comment:
you can avoid the exception handling (which is always dangerous to catch stuff unconditionally) by asking the registry embedded in `self.env.registry`, and condense everything to

```python
self -= self.filtered(lambda x: x.model not in self.env.registry)
```